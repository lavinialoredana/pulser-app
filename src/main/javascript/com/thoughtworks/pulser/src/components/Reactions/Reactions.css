.Reactions-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    margin:3rem;
}