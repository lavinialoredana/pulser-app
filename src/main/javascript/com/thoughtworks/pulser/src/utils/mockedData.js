export const mockedDataArray = [
  {
    id: "63c8121120b26e6a06d92c19",
    face: "AWESOME",
    body: "having a great time!",
  },
  {
    id: "63c8122320b26e6a06d92c1a",
    face: "HAPPY",
    body: "this is really nice",
  },
  {
    id: "63c8125220b26e6a06d92c1b",
    face: "SAD",
    body: "well... not so good",
  },
  {
    id: "63c8128320b26e6a06d92c1c",
    face: "AWFUL",
    body: "not good at all, not gonna lie",
  },
];
