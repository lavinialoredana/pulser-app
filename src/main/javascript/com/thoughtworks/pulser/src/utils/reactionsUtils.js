import awesomeFace from "../assets/images/awesome_face.png";
import happyFace from "../assets/images/happy_face.png";
import sadFace from "../assets/images/sad_face.png";
import awfulFace from "../assets/images/awful_face.png";

export const reactionsArray = [
  {
    key: "AWESOME",
    reactionName: "Awesome",
    reactionImage: awesomeFace,
    reactionAlt: "awesome_face",
  },
  {
    key: "HAPPY",
    reactionName: "Happy",
    reactionImage: happyFace,
    reactionAlt: "happy_face",
  },
  {
    key: "SAD",
    reactionName: "Sad",
    reactionImage: sadFace,
    reactionAlt: "sad_face",
  },
  {
    key: "AWFUL",
    reactionName: "Awful",
    reactionImage: awfulFace,
    reactionAlt: "awful_face",
  },
];
