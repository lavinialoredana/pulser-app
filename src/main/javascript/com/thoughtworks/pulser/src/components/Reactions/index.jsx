import React from "react";
import FeedbackReaction from "../FeedbackReaction";
import { reactionsArray } from "../../utils/reactionsUtils";
import "./Reactions.css";

const Reactions = ({ onReactionChange, onReactionPress }) => {
  return (
    <div className="Reactions-container">
      {reactionsArray.map((reaction) => {
        return (
          <div key={reaction.key}>
            <FeedbackReaction
              reactionName={reaction.reactionName}
              reactionImage={reaction.reactionImage}
              reactionAlt={reaction.reactionAlt}
              reactionKey={reaction.key}
              onReactionClick={onReactionChange}
              isReactionPressed={onReactionPress}
            />
          </div>
        );
      })}
    </div>
  );
};

export default Reactions;
