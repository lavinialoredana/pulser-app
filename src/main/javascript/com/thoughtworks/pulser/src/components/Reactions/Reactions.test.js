import { cleanup, render, screen } from "@testing-library/react";
import Reactions from "./index.jsx";
import { reactionsArray } from "../../utils/reactionsUtils";

describe("Reactions Component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should match snapshot", () => {
    const onReactionPressMock = jest.fn();
    const { asFragment } = render(
      <Reactions
        reactions={reactionsArray}
        onReactionPress={onReactionPressMock}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should contain Awesome title and awesome_face image", () => {
    const onReactionPressMock = jest.fn();
    render(
      <Reactions
        reactions={reactionsArray}
        onReactionPress={onReactionPressMock}
      />
    );

    const titleText = screen.getByText("Awesome");
    const awesomeImage = screen.getByAltText("awesome_face");

    expect(titleText).toBeInTheDocument();
    expect(awesomeImage.src).toContain("awesome_face.png");
  });

  it("should render Happy title and happy_face image", () => {
    const onReactionPressMock = jest.fn();

    render(
      <Reactions
        reactions={reactionsArray}
        onReactionPress={onReactionPressMock}
      />
    );

    const titleText = screen.getByText("Happy");
    const happyImage = screen.getByAltText("happy_face");

    expect(titleText).toBeInTheDocument();
    expect(happyImage.src).toContain("happy_face.png");
  });

  it("should render Sad title and sad_face image", () => {
    const onReactionPressMock = jest.fn();

    render(
      <Reactions
        reactions={reactionsArray}
        onReactionPress={onReactionPressMock}
      />
    );

    const titleText = screen.getByText("Sad");
    const awesomeImage = screen.getByAltText("sad_face");

    expect(titleText).toBeInTheDocument();
    expect(awesomeImage.src).toContain("sad_face.png");
  });

  it("should render Awful title and awful_face image", () => {
    const onReactionPressMock = jest.fn();
    render(
      <Reactions
        reactions={reactionsArray}
        onReactionPress={onReactionPressMock}
      />
    );

    const titleText = screen.getByText("Awful");
    const awesomeImage = screen.getByAltText("awful_face");

    expect(titleText).toBeInTheDocument();
    expect(awesomeImage.src).toContain("awful_face.png");
  });
});
