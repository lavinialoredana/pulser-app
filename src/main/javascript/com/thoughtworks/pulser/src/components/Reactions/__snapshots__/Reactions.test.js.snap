// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reactions Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Reactions-container"
    data-testid="reactions"
  >
    <div>
      <div
        class="Feedback-reaction-component  "
        data-testid="feedback-reaction"
      >
        <h3
          data-testid="feedBackReaction-title"
        >
           Awesome
        </h3>
        <img
          alt="awesome_face"
          class="Feedback-reaction-image"
          src="awesome_face.png"
        />
      </div>
    </div>
    <div>
      <div
        class="Feedback-reaction-component  "
        data-testid="feedback-reaction"
      >
        <h3
          data-testid="feedBackReaction-title"
        >
           Happy
        </h3>
        <img
          alt="happy_face"
          class="Feedback-reaction-image"
          src="happy_face.png"
        />
      </div>
    </div>
    <div>
      <div
        class="Feedback-reaction-component  "
        data-testid="feedback-reaction"
      >
        <h3
          data-testid="feedBackReaction-title"
        >
           Sad
        </h3>
        <img
          alt="sad_face"
          class="Feedback-reaction-image"
          src="sad_face.png"
        />
      </div>
    </div>
    <div>
      <div
        class="Feedback-reaction-component  "
        data-testid="feedback-reaction"
      >
        <h3
          data-testid="feedBackReaction-title"
        >
           Awful
        </h3>
        <img
          alt="awful_face"
          class="Feedback-reaction-image"
          src="awful_face.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
